apps/web/pages/api/verify-booking-token.ts

```diff
@@ -0,0 +1,96 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import { defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import { UserPermissionRole } from "@calcom/prisma/enums";
+import { TRPCError } from "@calcom/trpc/server";
+import { createContext } from "@calcom/trpc/server/createContext";
+import { bookingsRouter } from "@calcom/trpc/server/routers/viewer/bookings/_router";
+import { createCallerFactory } from "@calcom/trpc/server/trpc";
+import type { UserProfile } from "@calcom/types/UserProfile";
+
+enum DirectAction {
+  ACCEPT = "accept",
+  REJECT = "reject",
+}
+
+const querySchema = z.object({
+  action: z.nativeEnum(DirectAction),
+  token: z.string(),
+  bookingUid: z.string(),
+  userId: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse<Response>) {
+  const { action, token, bookingUid, userId } = querySchema.parse(req.query);
+  const { reason } = z.object({ reason: z.string().optional() }).parse(req.body);
+
+  const booking = await prisma.booking.findUnique({
+    where: { oneTimePassword: token },
+  });
+
+  if (!booking) {
+    res.redirect(`/booking/${bookingUid}?error=${encodeURIComponent("Error confirming booking")}`);
+    return;
+  }
```

Comment:
@zomars why do we remove the 404 page?

Reply:
I think we can leave it like that for now.

Reply:
Also the 404 page didn't work as expected as it might conflict with UID